source/grammar/qasm3.g4

```diff
@@ -276,34 +279,73 @@ logicalOperator
     | '||'
     ;
 
+shiftOperator
+    : '<<'
+    | '>>'
+    ;
+
+additiveOperator
+    : '+'
+    | '-'
+    ;
+
+multiplicativeOperator
+    : '*'
+    | '/'
+    | '%'
+    ;
+
+powerOperator
+    : '**'
+    ;
+
 expressionStatement
     : expression SEMICOLON
     ;
 
 expression
     // include terminator/unary as base cases to simplify parsing
     : expressionTerminator
-    | unaryExpression
```

Comment:
I have these base cases to simplify the parse tree--see the test results you generated vs the ones that were there before. Your parse trees are much longer are more complicated. If we have a terminator or a unary expr, we don't necessarily need to go through the entire expression hierarchy